plugins/modules/systemd_creds_decrypt.py

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright (c) 2024, Ansible Project
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+module: systemd_creds_decrypt
+short_description: Systemd's systemd-creds decrypt plugin
+description:
+    - This module decrypts input using systemd's systemd-creds decrypt.
+author:
+    - Thomas Sjögren (@konstruktoid)
+version_added: '10.2.0'
```

Comment:
Every module needs to declare the standard attributes. I think for these modules similar attributes than for `_info` modules fit:
```suggestion
version_added: '10.2.0'
extends_documentation_fragment:
  - community.general.attributes
attributes:
  check_mode:
    support: full
    details:
      - This action does not modify state.
  diff_mode:
    support: N/A
    details:
      - This action does not modify state.
```
In fact I'm wondering whether these modules shouldn't be `_info` modules, since they do not manage state, but retrieve information (encrypted/decrypted text, given some input).

Reply:
The difference I would say is that the `_info` modules most often doesn't actually generate or convert anything, they "just" retrieve things, while these modules encrypt/decrypt strings and generate a output based on those options.

But it could all be just semantics.

Reply:
Encrypting/decrypting is definitely a gray area, if no files on disk are involved. For decryption _info would fit better than for encryption, but in both cases not using _info should also be fine. (A somewhat related example is the https://docs.ansible.com/ansible/devel/collections/community/crypto/openssl_signature_info_module.html module, as a counterpart of the https://docs.ansible.com/ansible/devel/collections/community/crypto/openssl_signature_module.html module.)

In any case, I'm fine with keeping the current names :)

Reply:
Maybe they should be filters instead? (Chipping in late in this discussion)

Reply:
That's even a worse gray area ;-) In any case, a module is better suited since the encryption/decryption here is machine specific, so it has to happen on the remote system. If it wouldn't be machine specific, it could also happen on the controller.